flowgate-api/src/main/java/com/vmware/flowgate/controller/FacilitySoftwareController.java

```diff
@@ -184,18 +183,71 @@ public void updateFacilitySoftwareConfig(@RequestBody FacilitySoftwareConfig con
          throw WormholeRequestException.NotFound("FacilitySoftwareConfig", "id", config.getId());
       }
       FacilitySoftwareConfig old = oldServerOptional.get();
+      decryptServerPassword(old);
+
+      if (!checkFacilitySoftwareConfigIsUpdate(config, old)) {
```

Comment:
it doesn't need to do those check. just simply apply the new value. since this API will not used a lot. also this will make it buggy in then feature when there are new fields added to the advance part.